internal/services/containers/container_registry_cache_rule_data_source.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package containers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/containerregistry/2021-08-01-preview/registries"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/containerregistry/2023-07-01/cacherules"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type ContainerRegistryCacheRuleDataSource struct{}
+
+func (ContainerRegistryCacheRuleDataSource) ResourceType() string {
+	return "azurerm_container_registry_cache_rule"
+}
+
+func (ContainerRegistryCacheRuleDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Containers.ContainerRegistryClient_v2023_07_01.CacheRules
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
```

Comment:
When we have the model struct for the data source defined below we can decode it here 
```suggestion

			var state ContainerRegistryCacheRuleDataSourceModel
			if err := metadata.Decode(&state); err != nil {
				return err
			}

```